swri_roscpp/include/swri_roscpp/topic_service_client.h

```diff
@@ -64,7 +63,12 @@ class TopicServiceClientRaw
                 const std::string &service,
                 const std::string &client_name = "")
   {
-    name_ = client_name.length() ? client_name : ros::this_node::getName();
+    int random_int = rand()%10000;
```

Comment:
I'd rather not use a random number from 0 to 9999 for this since there's a possibility of a collision (especially if you're not seeding the RNG).  If the purpose of this is just to have a unique name, adding a UUID would be more reliable; if it's useful for this to be a relatively small number, I'd rather have something like a static variable that gets incremented every time it's used.

Reply:
We tried to do that, but ran into the issue where we couldn't figure out where to put a static variable

Reply:
We might have to do the UUID approach.

Reply:
You totally nerd-sniped me on this one.

```c++
struct MonotonicCounter
{
  static uint64_t next()
  {
    ++value_;
    return value_;
  }

  private:
  static uint64_t value_ = 0;
}
```

Usage
```c++
if (client_name.empty()
{
  std::ostringstream ss;
  ss << ros::this_node::getName()  << MonotonicCounter::next();
  name_ = ss.str()
}
else
{
  name_ = client_name;
}
```

Reply:
:(
`swri_roscpp/include/swri_roscpp/topic_service_client.h:49:28: error: ISO C++ forbids in-class initialization of non-const static member ‘swri::MonotonicCounter::value_’
   static uint64_t value_ = 0;`

Reply:
C++03 makes static initialization hard.